cmd/zt_test.go

```diff
@@ -73,15 +73,20 @@ const (
 // Will truncate the end of the test name, if there is not enough room for it, followed by the time-based suffix,
 // with a non-zero maxLen.
 func generateName(prefix string, maxLen int) string {
-	// These next lines up through the for loop are obtaining and walking up the stack
-	// trace to extrat the test name, which is stored in name
-	pc := make([]uintptr, 10)
-	runtime.Callers(0, pc)
-	f := runtime.FuncForPC(pc[0])
-	name := f.Name()
-	for i := 0; !strings.Contains(name, "Suite"); i++ { // The tests are all scoped to the suite, so this ensures getting the actual test name
-		f = runtime.FuncForPC(pc[i])
-		name = f.Name()
+	// The following lines step up the stack find the name of the test method
+	// Note: the way to do this changed in go 1.12, refer to release notes for more info
+	var pcs [10]uintptr
+	n := runtime.Callers(1, pcs[:])
+	frames := runtime.CallersFrames(pcs[:n])
+	name := "TestFoo" // default stub "Foo" is used if anything goes wrong with this procedure
+	for {
```

Comment:
Yeah breaking is probably a better way to do this than what we've got in azure-storage-blob-go. We should get that changed over there when we find a convenient time.

Reply:
Sounds good. The next change we do there can include this.

Reply:
Actually, this is really silly in comparison to what we've got in azblob.

```for f, next := frames.Next(); next; f, next = frames.Next() {
		name = f.Function
		if strings.Contains(name, "Suite") {
			break
		}
	}```

Consider this version of the loop over in azblob.

Reply:
We're doing roughly the exact same thing, just this one has an exit condition inside the loop itself, rather than in the initialization of the for loop.